components/friend.php

```diff
@@ -0,0 +1,337 @@
+<?php
+/**
+ * Manage BuddyPress Friends.
+ *
+ * @since 1.6.0
+ */
+class BPCLI_Friend extends BPCLI_Component {
+
+	/**
+	 * Object fields.
+	 *
+	 * @var array
+	 */
+	protected $obj_fields = array(
+		'id',
+		'initiator_user_id',
+		'friend_user_id',
+		'is_confirmed',
+		'is_limited',
+	);
+
+	/**
+	 * Create a new friendship.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <initiator>
+	 * : ID of the user who is sending the friendship request. Accepts either a user_login or a numeric ID.
+	 *
+	 * <friend>
+	 * : ID of the user whose friendship is being requested. Accepts either a user_login or a numeric ID.
+	 *
+	 * [--force-accept=<force-accept>]
+	 * : Whether to force acceptance.
+	 * ---
+	 * default: false
+	 * ---
+	 *
+	 * [--silent=<silent>]
+	 * : Silent friendship creation.
+	 * ---
+	 * default: false
+	 * ---
+	 *
+	 * [--porcelain]
+	 * : Return only the friendship id.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp friend create user1 another_use
+	 *     Success: Friendship successfully created.
+	 *
+	 *     $ wp bp friend create user1 another_use --force-accept=true
+	 *     Success: Friendship successfully created.
+	 *
+	 * @alias add
+	 */
+	public function create( $args, $assoc_args ) {
+		$r = wp_parse_args( $assoc_args, array(
+			'force-accept' => false,
+			'silent'       => false,
+		) );
+
+		// Members.
+		$initiator = $this->get_user_id_from_identifier( $args[0] );
+		$friend = $this->get_user_id_from_identifier( $args[1] );
+
+		if ( ! $initiator || ! $friend ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		if ( ! friends_add_friend( $initiator->ID, $friend->ID, $r['force-accept'] ) ) {
```

Comment:
`friends_add_friend()` returns `true` if the users are already friends. I find this to be hugely confusing, but we probably can't change it in BP for backward compatibility reasons. I find it even more confusing in the CLI context, because the success message is `Friendship successfully created...`. Can we have a separate `friends_check_friendship()` check here, which spits out an error `These users are already friends` or something like that?